src/metricscollector/collector/collector.go

```diff
@@ -18,53 +17,60 @@ type NoaaConsumer interface {
 }
 
 type Collector struct {
-	conf     *config.CollectorConfig
-	logger   lager.Logger
-	cfc      cf.CfClient
-	noaa     NoaaConsumer
-	database db.DB
-	cclock   clock.Clock
-	doneChan chan bool
-	pollers  map[string]*AppPoller
-	ticker   clock.Ticker
-	lock     *sync.Mutex
+	conf         *config.CollectorConfig
+	logger       lager.Logger
+	cfc          cf.CfClient
+	noaa         NoaaConsumer
+	database     db.DB
+	cclock       clock.Clock
+	createPoller func(string, time.Duration, lager.Logger, cf.CfClient, NoaaConsumer, db.DB, clock.Clock) AppPoller
+	doneChan     chan bool
+	pollers      map[string]AppPoller
+	ticker       clock.Ticker
+	lock         *sync.Mutex
+}
+
+var createAppPollerFunc = func(appId string, pollInterval time.Duration, logger lager.Logger, cfc cf.CfClient, noaa NoaaConsumer, database db.DB, pclcok clock.Clock) AppPoller {
```

Comment:
Would have just called it createAppPoller.
You can always pass around a function by its name.